utils/economic-checks/src/targets.rs

```diff
@@ -0,0 +1,703 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use arbitrary::Unstructured;
+use codec::Encode;
+use common::ValueTree;
+use frame_support::dispatch::DispatchError;
+use gear_core::ids::ProgramId;
+use gear_runtime::{Gear, Origin, Runtime};
+use primitive_types::H256;
+use rand::{rngs::StdRng, seq::SliceRandom, RngCore, SeedableRng};
+use sp_core::sr25519;
+use sp_std::collections::btree_map::BTreeMap;
+use wasm_mutate::{ErrorKind, WasmMutate};
+use wasmparser::Validator;
+
+use std::fmt;
+
+use crate::{
+    contracts::{GENERAL_WASM_BINARY, MUL_CONST_WASM_BINARY, NCOMPOSE_WASM_BINARY},
+    util::*,
+    Params, MAX_QUEUE_LEN,
+};
+
+type TargetOutcome = Result<GasUsageStats, DispatchError>;
+
+struct Seed([u8; 32]);
+
+impl fmt::Display for Seed {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "0x{}", hex::encode(&self.0))
+    }
+}
+
+impl fmt::Debug for Seed {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(self, f)
+    }
+}
+
+const MAX_BLOCK: u16 = 20;
+const MIN_BLOCK: u16 = 3;
+const NUM_TRIES: u16 = 5;
+
+#[derive(Default, Debug)]
+pub struct GasUsageStats {
+    total_gas_supply: u64,
+    accounted_gas: u64,
+    total_balance: u128,
+    initial_balance: u128,
+    #[allow(dead_code)]
+    reserved_balance: u128,
+}
+
+impl GasUsageStats {
+    fn new(
+        total_gas: u64,
+        accounted_gas: u64,
+        total_balance: u128,
+        initial_balance: u128,
+        reserved_balance: u128,
+    ) -> Self {
+        Self {
+            total_gas_supply: total_gas,
+            accounted_gas,
+            total_balance,
+            initial_balance,
+            reserved_balance,
+        }
+    }
+}
+
+pub fn composer_target(params: &Params) -> TargetOutcome {
+    let alice = get_account_id_from_seed::<sr25519::Public>("Alice");
+    let (mut ext, pool) = with_offchain_ext(
+        vec![
+            (alice.clone(), 1_000_000_000_000_000_u128),
+            (
+                // Just in case, to avoid removing account as dust
+                get_account_id_from_seed::<sr25519::Public>("Val"),
+                1_000_u128,
```

Comment:
ahh, got from where this 1_000 appeared
isn't better than to somehow throw runtime setting for it?

if we gonna increase existential deposit, this will fail